File: sdk/java/src/test/java/com/flagent/client/model/CreateConstraintRequestTest.java
package com.flagent.client.model;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

/**
 * Model tests for CreateConstraintRequest
 */
class CreateConstraintRequestTest {

    @Test
    void testCreateConstraintRequest() {
        CreateConstraintRequest model = new CreateConstraintRequest()
            .property("region")
            .operator("IN")
            .value("EU,US");
        Assertions.assertNotNull(model);
        Assertions.assertEquals("region", model.getProperty());
        Assertions.assertEquals("IN", model.getOperator());
        Assertions.assertEquals("EU,US", model.getValue());
    }

    @Test
    void propertyTest() {
        CreateConstraintRequest model = new CreateConstraintRequest().property("tier");
        Assertions.assertEquals("tier", model.getProperty());
    }

    @Test
    void operatorTest() {
        CreateConstraintRequest model = new CreateConstraintRequest().operator("EQ");
        Assertions.assertEquals("EQ", model.getOperator());
    }

    @Test
    void valueTest() {
        CreateConstraintRequest model = new CreateConstraintRequest().value("premium");
        Assertions.assertEquals("premium", model.getValue());
    }

}

File: sdk/dart/test/create_constraint_request_test.dart
import 'package:test/test.dart';
import 'package:flagent_client/flagent_client.dart';

// tests for CreateConstraintRequest
void main() {
  final instance = CreateConstraintRequest((b) => b
    ..property = 'tier'
    ..operator_ = 'EQ'
    ..value = 'premium');

  group(CreateConstraintRequest, () {
    test('to test the property `property`', () async {
      expect(instance.property, equals('tier'));
    });

    test('to test the property `operator_`', () async {
      expect(instance.operator_, equals('EQ'));
    });

    test('to test the property `value`', () async {
      expect(instance.value, equals('premium'));
    });
  });
}
